### PR TITLE
Update dependencies for 5.4.0 and fix some help messages

### PR DESCRIPTION
**Pull request by str4d**
Includes changes backported from the following upstream PRs (in order to pull in fixes to Boost deprecations):
- bitcoin/bitcoin#7290
- bitcoin/bitcoin#11131
- bitcoin/bitcoin#12305
- bitcoin/bitcoin#12797

### CODE REVIEW

File: src/init.cpp [hunk 1]
```diff
@@ -441,8 +441,11 @@ std::string HelpMessage(HelpMessageMode mode)
     strUsage += HelpMessageOpt("-debugmetrics", _("Include debug metrics in exposed node metrics."));
 
     strUsage += HelpMessageGroup(_("Debugging/Testing options:"));
+    strUsage += HelpMessageOpt("-uacomment=<cmt>", _("Append comment to the user agent string"));
     if (showDebug)
     {
+        strUsage += HelpMessageOpt("-checkblockindex", strprintf("Do a full consistency check for mapBlockIndex, setBlockIndexCandidates, chainActive and mapBlocksUnlinked occasionally. Also sets -checkmempool. (default: %u)", Params().DefaultConsistencyChecks()));
```
[the remark this answers is not in the document]
Comment (daira) [reply src/init.cpp]: This appears to have been an error in the Bitcoin message; see https://github.com/bitcoin/bitcoin/pull/12797 .

[the remark this answers is not in the document]
Comment (str4d) [reply src/init.cpp]: Indeed, it looks like this was fixed-up in https://github.com/bitcoin/bitcoin/pull/12797. I'll backport that commit as well.

File: src/init.cpp [hunk 2]
```diff
@@ -465,11 +468,14 @@ std::string HelpMessage(HelpMessageMode mode)
...
         _("For multiple specific categories use -debug=<category> multiple times."));
     strUsage += HelpMessageOpt("-experimentalfeatures", _("Enable use of experimental features"));
+    if (showDebug)
+        strUsage += HelpMessageOpt("-nodebug", "Turn off debugging messages, same as -debug=0");
     strUsage += HelpMessageOpt("-help-debug", _("Show all debugging options (usage: --help -help-debug)"));
     strUsage += HelpMessageOpt("-logips", strprintf(_("Include IP addresses in debug output (default: %u)"), DEFAULT_LOGIPS));
     strUsage += HelpMessageOpt("-logtimestamps", strprintf(_("Prepend debug output with timestamp (default: %u)"), DEFAULT_LOGTIMESTAMPS));
     if (showDebug)
     {
+        strUsage += HelpMessageOpt("-mocktime=<n>", "Replace actual time with <n> (default: 0)");
```
Comment (daira) [inline src/init.cpp]: If we document `-mocktime` then we should also document `-clockoffset`, which we added in #6042. (Note that they cannot be used together.)

**Comment by str4d** [comment]
[Force-pushed](https://github.com/zcash/zcash/compare/018345523f344e76e372c383abd1a6c3cc55c7e0..0de7a08c86cadb47b9fb3adfffb31feb401e97fe) to backport bitcoin/bitcoin#12797.

**Comment by daira** [comment]
The commit message for 145d259e3e83ad3f6290d65398e8d1019da11466 references `bitcoin.conf` and `~/.bitcoin`; change these to `zcash.conf` and `~/.zcash`.

**Comment by str4d** [comment]
I normally don't modify the commit messages of cherry-picked commits, only add to them. I don't think it matters much; the real filename is set from `BITCOIN_CONF_FILENAME` which is set to `zcash.conf`.

**Review by daira — DISMISSED** [review]
utACK. Consider documenting `-clockoffset`.

**Comment by daira** [comment]
utACK 7e420417728816e83834978cd53ccbbfa0a410d9

**Review by daira — APPROVED** [review]

**Pull request merged**